test/common/event/scaled_range_timer_manager_test.cc

```diff
@@ -0,0 +1,450 @@
+#include <chrono>
+
+#include "envoy/event/timer.h"
+
+#include "common/event/scaled_range_timer_manager.h"
+
+#include "test/mocks/common.h"
+#include "test/mocks/event/wrapped_dispatcher.h"
+#include "test/test_common/simulated_time_system.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Event {
+namespace {
+
+using testing::_;
+using testing::ElementsAre;
+using testing::InSequence;
+using testing::IsEmpty;
+using testing::Mock;
+using testing::MockFunction;
+using testing::NiceMock;
+using testing::StrictMock;
+
+class ScopeTrackingDispatcher : public WrappedDispatcher {
```

Comment:
Yes, my idea with the onFatalError extensions would be to invoke them here: https://github.com/envoyproxy/envoy/blob/12a103cb4bd667208cd5fba9b3bf29defe2c1dfe/source/common/event/dispatcher_impl.h#L88

So it currently wouldn't need a public accessor. 

An accessor would get tricky if you try to add an accessor that wasn't running on the same thread, and I don't think it would be a good idea.